src/compiler/checker.ts

```diff
@@ -22005,6 +22005,19 @@ namespace ts {
                     }
                 }
             }
+
+            // If location is stringLiteral, it might be argument in ElementAccessExpression, like a["Q"].
+            // Deal it just like identifier/PropertyAccessExpression.

```

Comment:
```suggestion
            // If the location is a StringLiteral, it might be the argument to an ElementAccessExpression (e.g. like foo["bar"])
            // Deal it the same way as we would with an Identifier or PropertyAccessExpression.
```

Reply:
Strange, why could I not use `Commit suggestion`? It always says `This diff has recently been updated.`